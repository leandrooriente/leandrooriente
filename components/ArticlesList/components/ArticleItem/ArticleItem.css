.wrapper {
  margin-bottom: 25px;
}

.wrapper:last-child {
  margin-bottom: 0;
}

.title {
  line-height: 1.6;
  margin: 0 0 10px;
}

.link {
  margin: 0 -5px;
  padding: 0 5px;
  display: inline-block;
  color: var(--primary-color);
  text-decoration: none;
  border-bottom: 1px solid var(--light-gray);
  transition: all var(--fast-transition) ease-in;  
}

.link:hover {
  background: var(--primary-color);
  color: var(--light-gray);
}

.badge {
  margin: 0 10px 10px 0;
}

.badge:last-child {
  margin: 0;
}