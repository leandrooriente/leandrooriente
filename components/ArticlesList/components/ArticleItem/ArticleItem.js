import React from 'react';
import PropTypes from 'prop-types';
import formatArticleDate from 'helpers/formatArticleDate';
import Badge from 'components/Badge';
import Link from 'next/link';
import ArticleShape from '../../propTypes/article';
import css from './ArticleItem.css';

const ArticleItem = ({ article }) => (
  <div className={css.wrapper}>
    <p className={css.title}>
      <Link href={article.url}>
        <a className={css.link} href={article.url}>
          {article.title}
        </a>
      </Link>
    </p>
    <div>
      <Badge>{formatArticleDate(article.date)}</Badge>
      <Badge>{article.lang}</Badge>
    </div>
  </div>
);

ArticleItem.propTypes = {
  article: PropTypes.shape(ArticleShape).isRequired,
};

export default ArticleItem;
