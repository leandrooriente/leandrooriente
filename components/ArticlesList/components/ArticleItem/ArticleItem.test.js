import React from 'react';
import { mount } from 'enzyme';
import articlesMock from '__mocks__/articles';
import ArticleItem from './ArticleItem';

describe('<ArticleItem />', () => {
  it('should render <ArticleItem /> component', () => {
    const wrapper = mount(
      <ArticleItem article={articlesMock[0]} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
