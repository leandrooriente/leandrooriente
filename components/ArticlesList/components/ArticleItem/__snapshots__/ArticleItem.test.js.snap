// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleItem /> should render <ArticleItem /> component 1`] = `
<ArticleItem
  article={
    Object {
      "author": "Leandro Oriente",
      "date": "2014-03-09T00:00:00.000Z",
      "description": "Tracei uma nova meta de vida. Aprender inglês de verdade em 1 ano. Tenho uma base razoavel a bastante tempo que me permite ler artigos, posts, documentações em inglês e compreender boa parte delas mas que não é o suficiente para conversar por 5 minutos com alguém fluente sem parecer um idiota.",
      "html": null,
      "lang": "pt-BR",
      "slug": "como-aprender-ingles",
      "thumbnail": "/images/joel-santana-thumb.jpg",
      "title": "Como aprender inglês",
      "url": "/como-aprender-ingles",
    }
  }
>
  <div
    className="wrapper"
  >
    <p
      className="title"
    >
      <Link
        href="/como-aprender-ingles"
      >
        <a
          className="link"
          href="/como-aprender-ingles"
          onClick={[Function]}
        >
          Como aprender inglês
        </a>
      </Link>
    </p>
    <div>
      <Badge
        className="badge"
      >
        <span
          className="badge badge"
        >
          March 2014
        </span>
      </Badge>
      <Badge
        className="badge"
      >
        <span
          className="badge badge"
        >
          pt-BR
        </span>
      </Badge>
    </div>
  </div>
</ArticleItem>
`;
